aidial_adapter_bedrock/llm/model/adapter.py

```diff
@@ -106,23 +111,31 @@ async def get_bedrock_adapter(
                 await Bedrock.acreate(aws_client_config), model
             )
         case (
-            ChatCompletionDeployment.META_LLAMA2_13B_CHAT_V1
-            | ChatCompletionDeployment.META_LLAMA2_70B_CHAT_V1
+            ChatCompletionDeployment.META_LLAMA3_8B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_70B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_1_8B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_2_1B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_2_3B_INSTRUCT_V1
         ):
-            return MetaAdapter.create(
-                await Bedrock.acreate(aws_client_config), model, llama2_config
+            return ConverseAdapter(
+                deployment=model,
+                bedrock=await Bedrock.acreate(aws_client_config),
+                storage=create_file_storage(api_key),
+                input_tokenizer_factory=llama_tokenizer_factory,
+                support_tools=False,
             )
         case (
-            ChatCompletionDeployment.META_LLAMA3_8B_INSTRUCT_V1
-            | ChatCompletionDeployment.META_LLAMA3_70B_INSTRUCT_V1
+            ChatCompletionDeployment.META_LLAMA3_1_70B_INSTRUCT_V1
             | ChatCompletionDeployment.META_LLAMA3_1_405B_INSTRUCT_V1
-            | ChatCompletionDeployment.META_LLAMA3_1_70B_INSTRUCT_V1
-            | ChatCompletionDeployment.META_LLAMA3_1_8B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_2_11B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_2_90B_INSTRUCT_V1
         ):
-            return MetaAdapter.create(
-                await Bedrock.acreate(aws_client_config),
-                model,
-                llama3_config,
+            return ConverseStreamingEmulateAdapter(
+                deployment=model,
+                bedrock=await Bedrock.acreate(aws_client_config),
+                storage=create_file_storage(api_key),
+                input_tokenizer_factory=llama_tokenizer_factory,
+                support_tools=True,
```

Comment:
Maybe add another feature flag: `support_tools_in_streaming_mode` and avoid multiple classes?

Reply:
I intentionally avoided adding such parameter in parent class

I think we should stick to single-responsibility principle from SOLID here:
It's better to have a lot of specialized classes, than one big swiss-knife class

This functionality (emulation of stream if we receive tools) - doesn't seem like a common case for all models, so let's leave it inside of such specialized class

Reply:
ok, however, this leaves as with the naming which is quite hard to understand: `ConverseStreamingEmulateAdapter`, especially because it has an indefinite verb, which is usually interpreted in namings as a command.

Maybe `ConverseAdapterWithLimitedStreamingSupport` or `ConverseAdapterWithStreamingEmulation`?

Reply:
Renamed to `ConverseAdapterWithStreamingEmulation`